byway/src/main/webapp/destinations.js

```diff
@@ -0,0 +1,207 @@
+        origin: new google.maps.Point(0, 0),
+        anchor: new google.maps.Point(17, 34),
+        scaledSize: new google.maps.Size(25, 25)
+      };
+      // Create a marker for each place.
+      markers.push(
+        new google.maps.Marker({
+          map,
+          icon,
+          title: place.name,
+          position: place.geometry.location
+        })
+      );
+
+      if (place.geometry.viewport) {
+        // Only geocodes have viewport.
+        bounds.union(place.geometry.viewport);
+      } else {
+        bounds.extend(place.geometry.location);
+      }
+    });
+    map.fitBounds(bounds);
+}
+
+/** 
+* fetches start location and destinations from DestinationsServlet and adds to DOM
+*/
+function updateLocations(locationData){
+    document.getElementById('start-location').innerText = "Start Location :" + locationData.start;
+    const container = document.getElementById('destinations-container');
+    container.innerText = "Destinations:";
+    let destinationArray= locationData.destinations;
+    destinationArray.forEach((destination) => {
+      let destinationToAdd = document.createElement('p');
+      destinationToAdd.innerText = destination;
+      container.appendChild(destinationToAdd);
+    }) 
+}
+
+/**  
+* fills Start location Searchbox with previously input
+*/
+function updateStartDestination(locationData){
+    if (locationData.start == null){
+      document.getElementById('start-search-box').value = "";
+    }
+    else{
+      document.getElementById('start-search-box').value = locationData.start;
+    }
+}
+
+/**
+* fetches data from servlet
+*/
+function fetchDestinations(){
+  let promise= fetch('/api/destinations').then(response => response.json());
+  return promise;
+}
+
+/** 
+* Gets users current location 
+*/
+function getCurrentAddress(){
+  const geocoder = new google.maps.Geocoder();
+  geocoder.geocode({ 'location': userlatlng}, (results, status) => {
+    if (status === "OK") {
+      if (results[0]) {
+          console.log(results[0].formatted_address);
+        document.getElementById("start-search-box").value=results[0].formatted_address;
+      } else {
+          window.alert("No results found");
+        }
+    } else {
+        window.alert("Geocoder failed due to: " + status);
+      }
+  });
+}
+
+/** 
+* add event listener for submit button
+*/
+window.onload = function(){
+  document.getElementById('user-input-form').addEventListener('submit', (event) => {
+    event.preventDefault();
+    const formData = new  FormData(document.getElementById("user-input-form"));
+    fetch('/api/destinations', {method: 'POST', body:formData}).then((response)=>
+        response.json()).then(locationData => {
+        updateLocations(locationData);
+        updateStartDestination(locationData);
+      });
+    });
+}
+
+
+if (document.readyState === 'loading') {  // Loading hasn't finished yet
+  document.addEventListener('DOMContentLoaded', initializeDestinationsPage)
+} 
```

Comment:
It the page is already loaded, you probably want to call initializeDestinationsPage immediately, otherwise there's a chance that the page won't get initialized if the document loads before this script is executed and the event handler is added.

Reply:
added